packages/mui-material/src/Button/Button.js

```diff
@@ -398,22 +569,30 @@ const Button = React.forwardRef(function Button(inProps, ref) {
     </ButtonEndIcon>
   );
 
+  const loader = (
+    <ButtonLoadingIndicator className={classes.loadingIndicator} ownerState={ownerState}>
+      {loading && loadingIndicator}
+    </ButtonLoadingIndicator>
+  );
```

Comment:
Please, let's make this conditional. I don't think this can fly:

<img width="573" alt="SCR-20241119-ortp" src="https://github.com/user-attachments/assets/d7666ef7-8bb6-48a4-b56e-e2224b1c84e5">

https://deploy-preview-42987--material-ui.netlify.app/material-ui/react-button/#basic-button

Reply:
If it's conditional, it won't fix https://github.com/mui/material-ui/pull/35198.

Reply:
> I don't think this can fly

May I ask why? what's the issue of it?